File: From Sessions and Others/Vjudge/Practice 1/F.py
for i in range(1, int(input()) + 1):
    first, second = map(int, input().split())
    n, m = map(int, input().split())
    
    coins = min(first * (m * (n // (m + 1))) + min(first, second) * (n % (m + 1)), min(first, second) * n)
    
    print(coins)

File: From Sessions and Others/Vjudge/Practice 1/F.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    long long t, first, second, coins, n, m;

    cin >> t;

    while (t--) {
        cin >> first >> second;
        cin >> n >> m;

        coins = min(first * (m * (n / (m + 1))) + min(first, second) * (n % (m + 1)), min(first, second) * n);

        cout << coins << "\n";
    }

    return 0;
}